### PR TITLE
docs: runtime spec

### PR DESCRIPTION
**Pull request by Slesarew**

### CODE REVIEW

File: docs/runtime.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Simultaion runtime
+
+## About
+
+This document describes specifications of a main thread of a single simulation with given set of parametes.
```
Comment (Zhenya-Barannik) [inline docs/runtime.md]: parametes -> parameters

File: docs/runtime.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Every simulation is done in a single-threaded mode to preserve deterministic entropy for analysis. Observational behavior included in modeling is not discussed here as the model is totally oblivious to the fact that it is being analyzed in any way (all observational operations MUST be clean for the model).
```
Comment (Zhenya-Barannik) [inline docs/runtime.md:7]: By clean you mean functionally pure? Ofc observation should not change model, for example new key-value pairs (even key->"0") should not be added to a resource dict on read access. Mutability control in Rust will make this simple.

File: docs/runtime.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Additionally, games should not match closely each other's frequency within phase stability time frame and on model's timeframe. Notable exception to this rule is modeling of natural cycles, like day/night or weekly cycle.
+
+### Game pool
+
+Games proposed by providers are collected and shuffled in single pool.
+
+### Players exchange
+
+Depending on every game rules, it is connected to a player exchange. Exchange could have complex rules on how to exclude players. Different roles may be connected to differetn pools. It is important that pools are generated before games execution, based on pre-existing information.
```
Comment (Zhenya-Barannik) [inline docs/runtime.md]: differetn -> different

File: docs/runtime.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Pools might have drainable and non-drainable behaviour and may be shared or not between games. They should probably have deterministic hash-id which would identify these for game engine.
+
+Some pools might have degenerate nature like 1-player initiator pool.
+
+## Execution
+
+The model runs following events for every **tick**:
+
+1. Propose games from all game providers and collect these in game pool.
+2. Shuffle game pool (this is important to avoid turn order texture)
```
Comment (Zhenya-Barannik) [inline docs/runtime.md:58]: Or should we come up with something else then shuffling? See question about conflicting games in https://github.com/Alzymologist/Kuvio-model/pull/4

File: docs/runtime.md (hunk 1 above)
Comment (Slesarew) [inline docs/runtime.md]:
```suggestion
This document describes specifications of a main thread of a single simulation with given set of parameters.
```

File: docs/runtime.md (hunk 4 above)
Comment (Zhenya-Barannik) [reply docs/runtime.md:58]: We decided to implement ordered agents and shuffling.

**Review by Zhenya-Barannik — APPROVED** [review]

File: docs/runtime.md (hunk 3 above)
Comment (Slesarew) [reply docs/runtime.md]: do suggestions!
```suggestion
Depending on every game rules, it is connected to a player exchange. Exchange could have complex rules on how to exclude players. Different roles may be connected to different pools. It is important that pools are generated before games execution, based on pre-existing information.
```

File: docs/runtime.md (hunk 2 above)
Comment (Slesarew) [reply docs/runtime.md:7]: exactly. Unless we start messing with heavy dyn stuff which we shouldn't

**Pull request merged**